packages/app/src/components/inspector/ColorPickerGradient.tsx

```diff
@@ -121,22 +106,67 @@ const SwatchesGrid = memo(function SwatchesGrid({
   );
 });
 
+const GridItem = memo(
+  React.forwardRef(
+    (
+      {
+        background,
+        handleClick,
+        onContextMenu,
+      }: {
+        background: string;
+        handleClick: () => void;
+        onContextMenu: (event: React.MouseEvent) => void;
+      },
+      forwardedRef: ForwardedRef<HTMLDivElement>,
+    ) => (
+      <Square
+        ref={forwardedRef}
+        background={background}
+        onContextMenu={onContextMenu}
+        onClick={handleClick}
+      />
+    ),
+  ),
+);
+
 interface Props {
-  swatchID?: string;
   gradientType: Sketch.GradientType;
   gradientAssets: Sketch.GradientAsset[];
-  onChange?: (gradient: Sketch.Gradient) => void;
   onCreate: () => void;
+  onChange?: (gradient: Sketch.Gradient) => void;
+  onRename: (id: string, name: string) => void;
+  onDelete: (id: string) => void;
 }
 
-export default memo(function ColorPickerSwatches({
-  swatchID,
+export default memo(function ColorPickerGradients({
   gradientType,
   gradientAssets,
   onChange,
   onCreate,
+  onRename,
+  onDelete,
 }: Props) {
   const [gradientLayout, setGradientLayout] = useState<SwatchLayout>('grid');
+  const [gradientId, setGradientId] = useState('');
+
+  const handleSelectMenuItem = useCallback(
```

Comment:
am I missing something here ? I am not sure why when I hover on an option the item doesn't highlight

Reply:
I tried it a bit. I think the highlighting issue is a bug in the library that happens when there's a popup open. We don't need to fix this